docs/gadget-devel/metadata.md

```diff
@@ -0,0 +1,9 @@
+---
+title: 'metadata file'
+weight: 300
+---
+
+TODO:
```

Comment:
The gadget metadata file is named `gadget.yaml` and is included in the OCI image. It is used by IG at run-time to find out what the gadget can do.

It is defined in the [GadgetMetadata](https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget/pkg/metadata/v1#GadgetMetadata) struct.

It can be generated using `ig image build --update-metadata`, see hello-world-gadget.md#Creating-a-metadata-file for more details.

Reply:
Thanks. I added something like that. I didn't add the link back to the hello-world as people reading this probably already went through the hello world guide and are looking for more information.